### ISSUE TITLE
Program does not want to win, and plays with the meat

### ISSUE THREAD

**Issue by alcalyn** [cgao3/benzene-vanilla-cmake#14]
TL:DR; I can get it work on commit `f88893`, but not on master.

I got similar bug as described as in other issues, like program is crashing, or does not want to finish the game by avoiding playing winning moves.

Can be reproduced with:

- run mohex
- run these commands:

``` bash
boardsize 11
play-game i8 h10 j7 j6 k11 f6 j11 g10 i11 e5 h11 e11 g11 j8 f11 c6 f10 g7 e10 b6 d11 i6 c11 h6 d10 f7 c10 d5 b11 f5 b10 f2 a11 j2 a10 f4 a9
genmove w
```

I expect him to play one of the few last move to win the game.

Actually, I get:

```
warning: **** MoHexSearch returned empty sequence!
warning: **** Returning random move!
```

Then it is returning a random move:
- sometimes it is a legal move and it's playing anywhere, like humiliating you and keeping you in a desperate mood "_see, I'm winning but I'm playing with you, my meat!_",
- sometimes the cell is already occupied so I get:

```
? illegal move:  white invalid (invalid)
```

I can get around this issue by building from an older version, on the commit `f888938bc7ab051a04034a9e19a31399a31d2429`.

This works as expected, the program plays the winning moves, and with the same commands:

```
info: Captured cells caused win! Removing...
info: Winning SC.
= k5
```

**Comment by cgao3** [comment]
this might be an issue from the new ICE pattern. If the old commit (the one you referred) works, perhaps you can just stick on with that one.
